build/mac/BUILD.gn

```diff
@@ -18,19 +21,29 @@ action("sign_app") {
 
   script = "//build/gn_run_binary.py"
   shell_script = "//brave/build/mac/sign_app.sh"
-  if (brave_channel == "") {
-    provisioning_profile = "//brave/build/mac/release.provisionprofile"
-  }
-  else {
-    provisioning_profile = "//brave/build/mac/${brave_channel}.provisionprofile"
-  }
 
+  deps = [
+    "//brave:chrome_app",
+    "//chrome/installer/mac"
+  ]
+
+  if (is_official_build) {
+    if (brave_channel == "") {
+      provisioning_profile = "//brave/build/mac/release.provisionprofile"
+    } else {
+      provisioning_profile = "//brave/build/mac/${brave_channel}.provisionprofile"
+    }
+  } else {
+    # an empty dummy file just to simplify things
+    provisioning_profile = "//brave/build/mac/dummy.provisionprofile"
```

Comment:
This logic is changing; is an empty dummy.provisionprofile OK for non-official builds? (have you tested this code-path?)

Reply:
this logic isn't changing, signing never worked for non-official builds. This is in fact the only code path I have tested because I can't test the dev/beta signing

Reply:
for dev signing we only pass a provision profile to simplify things with the number of arguments. It isn't actually used in the signing script when `--development` (below) is passed